### PR TITLE
Fixed security issues

### PR DESCRIPTION
**Pull request by vishwjeet-pawar03**
## Description

[Provide a brief description of the changes in this PR]

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/nodejs/apps/src/modules/auth/controller/userAccount.controller.ts [hunk 1]
```diff
@@ -205,6 +205,15 @@ export class UserAccountController {
...
         throw new BadRequestError('Email is required');
       }
+
+      if (typeof email !== 'string') {
+        throw new BadRequestError('Email must be a string');
+      }
+
+      if (email.length > 254) {
+        throw new BadRequestError('Email address is too long');
+      }
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply backend/nodejs/apps/src/modules/auth/controller/userAccount.controller.ts]: Fixed

File: backend/nodejs/apps/src/modules/enterprise_search/controller/es_controller.ts [hunk 2]
```diff
@@ -176,6 +180,15 @@ export const streamChat =
...
     const modelInfo = extractModelInfo(req.body);
 
+    // Validate query parameter for XSS and format specifiers
+    if (req.body.query && typeof req.body.query === 'string') {
+      validateNoXSS(req.body.query, 'query');
+      validateNoFormatSpecifiers(req.body.query, 'query');
+      
+    } else if (!req.body.query) {
+      throw new BadRequestError('Query is required');
+    }
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply backend/nodejs/apps/src/modules/enterprise_search/controller/es_controller.ts:190]: Fixed

File: backend/nodejs/apps/src/modules/user_management/controller/users.controller.ts [hunk 3]
```diff
@@ -322,33 +367,48 @@ export class UserController {
...
         user.designation = updateFields.designation;
       }
-      if (updateFields.mobile) {
+      if (updateFields.mobile !== undefined) {
         user.mobile = updateFields.mobile;
       }
-      if (updateFields.address) {
+      if (updateFields.address !== undefined) {
         user.address = updateFields.address;
       }
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply backend/nodejs/apps/src/modules/user_management/controller/users.controller.ts]: Fixed

File: backend/python/app/api/routes/entity.py [hunk 4]
```diff
@@ -140,6 +140,18 @@ async def get_teams(
     limit: int = Query(10, ge=1, le=100, description="Number of items per page")
 ) -> JSONResponse:
     """Get all teams for the current user's organization with pagination and search"""
+    # Validate search parameter for format string vulnerabilities
+    if search:
+        from app.utils.format_string_safety import validate_no_format_specifiers
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply backend/python/app/api/routes/entity.py]: Fixed

File: backend/python/app/connectors/api/router.py [hunk 5]
```diff
@@ -1919,7 +1931,23 @@ async def get_records(
...
+        from app.utils.safe_integer import (
+            safe_calculate_skip,
+            safe_calculate_total_pages,
+        )
+
+        # Validate page and limit are within safe bounds
+        if page < 1:
+            raise HTTPException(status_code=400, detail="Page must be greater than 0")
+        if limit < 1 or limit > 100:
+            raise HTTPException(status_code=400, detail="Limit must be between 1 and 100")
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply backend/python/app/connectors/api/router.py]: Fixed

File: backend/nodejs/apps/src/modules/knowledge_base/validators/validators.ts [hunk 6]
```diff
@@ -336,11 +337,42 @@ export const getAllRecordsSchema = z.object({
...
+        { message: 'Search parameter contains potentially dangerous content (HTML tags, scripts, or format specifiers are not allowed)' },
+      )
       .transform((val) => {
         if (!val) return undefined;
         const trimmed = val.trim();
-        if (trimmed.length > 100) {
-          throw new Error('Search term too long');
+        if (trimmed.length > 500) {
+          throw new Error('Search term too long (max 500 characters)');
         }
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply backend/nodejs/apps/src/modules/knowledge_base/validators/validators.ts]: Fixed

File: backend/nodejs/apps/src/modules/auth/controller/userAccount.controller.ts [hunk 7]
```diff
@@ -202,9 +202,7 @@ export class UserAccountController {
   ): Promise<void> {
     try {
       const { email } = req.body;
-      if (!email) {
```
Comment (abhishek-pipeshub) [inline backend/nodejs/apps/src/modules/auth/controller/userAccount.controller.ts:205]: why remove this check?

Comment (abhishek-pipeshub) [reply backend/nodejs/apps/src/modules/auth/controller/userAccount.controller.ts:205]: is it moved to validator?

File: backend/nodejs/apps/src/utils/xss-sanitization.ts [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+      // Remove script opening tags (complete and incomplete)
+      .replace(/<[\s]*script[\s]*[^>]{0,10000}>?/gi, '')
+      // Remove any remaining <script pattern
+      .replace(/<[\s]*script/gi, '')
+      // Remove iframe closing tags
+      .replace(/<\/[\s]*iframe[^>]{0,10000}>/gi, '')
+      // Remove iframe opening tags (complete and incomplete)
+      .replace(/<[\s]*iframe[\s]*[^>]{0,10000}>?/gi, '')
+      // Remove any remaining <iframe pattern
+      .replace(/<[\s]*iframe/gi, '')
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/nodejs/apps/src/utils/xss-sanitization.ts:251]: will add later
